v2/federation/fetch.go

```diff
@@ -15,19 +17,42 @@ var federationErr = errs.Class("federation")
 
 // FetchOption is an option used when dialing the bundle endpoint.
 type FetchOption interface {
-	apply(*fetchOptions)
+	apply(*fetchOptions) error
 }
 
 type fetchOptions struct {
-	transport *http.Transport
+	transport  *http.Transport
+	authMethod authMethod
 }
 
 // WithSPIFFEAuth authenticates the bundle endpoint with SPIFFE authentication
 // using the provided root store. It validates that the endpoint presents the
-// expected SPIFFE ID.
+// expected SPIFFE ID. This option cannot be used in conjuntion with WithWebPKIRoots
+// option.
 func WithSPIFFEAuth(bundleSource x509bundle.Source, endpointID spiffeid.ID) FetchOption {
-	return fetchOption(func(o *fetchOptions) {
+	return fetchOption(func(o *fetchOptions) error {
+		if o.authMethod != authMethodDefault {
+			return federationErr.New("authentication is already set to %v", o.authMethod)
```

Comment:
It might not be clear to the caller what they have done wrong. In particular the message they get back will be dependent on the order the options were provided in. Maybe for now, until we grow more options (if any), this message can simply be:
```suggestion
			return federationErr.New("cannot use both SPIFFE and Web PKI authentication")
```

Reply:
Much better, thanks